src/main/groovy/life/qbic/utils/NanoporeParser.groovy

```diff
@@ -156,13 +157,22 @@ class NanoporeParser {
      * @throws org.everit.json.schema.ValidationException
      */
     private static void validateJson(String json) throws ValidationException {
-        // Step1: load schema
+        // Step 1: load schema
         JSONObject jsonObject = new JSONObject(json)
-        InputStream schemaStream = OxfordNanoporeInstrumentOutput.getSchemaAsStream()
-        JSONObject rawSchema = new JSONObject(new JSONTokener(schemaStream))
-        Schema jsonSchema = SchemaLoader.load(rawSchema)
-        // Step2: validate against schema return if valid, throw exception if invalid
-        jsonSchema.validate(jsonObject)
+        try {
+            Schema jsonSchema = loadSchemaFromStream(OxfordNanoporeInstrumentOutput.getSchemaAsStream())
+            // Step 2: validate against schema return if valid, throw exception if invalid
+            jsonSchema.validate(jsonObject)
+        } catch (ValidationException validationException) {
+            // Step 2.5: validate against second schema type
+            Schema jsonSchema2 = loadSchemaFromStream(OxfordNanoporeInstrumentOutputV2.getSchemaAsStream())
+            jsonSchema2.validate(jsonObject)
+        }
+    }
```

Comment:
This could be made more explicit by making the multiple schema validations into a seperate method.

Reply:
just pushed a suggestion